ckanext/accessrequests/plugin.py

```diff
@@ -6,13 +6,41 @@
 from ckan.common import c, request, _
 from ckan.lib.activity_streams import activity_stream_string_functions
 get_action = logic.get_action
+import ckan.lib.helpers as h
+
 
 def activity_stream_string_reject_new_user(context, activity):
     return _("{actor} rejected new user {user}")
 
 def activity_stream_string_approve_new_user(context, activity):
     return _("{actor} approved new user {user}")
 
+def user_delete(context, data_dict=None):
+  """
+  :param context:
+  :return: True if user is sysadmin or admin in top level org
+  """
+  has_access = check_access_account_requests(context)
+  return {'success': True if has_access['success'] else False}
+
+
+def check_access_account_requests(context, data_dict=None):
+  """
+  :param context:
+  :return: True if user is sysadmin or admin in top level org
+  """
+  user = context.get('user')
+  orgs = logic.get_action('organization_list_for_user')({'user': user}, {'permission': 'admin'})
+  user_is_admin_in_top_org = None
+  if orgs:
+    for org in orgs:
+      group = model.Group.get(org['id'])
+      if group.id == (group.get_parent_group_hierarchy(type='organization') or [group])[0].id:
+        user_is_admin_in_top_org = True
+        break
+
+  return {'success': True if user_is_admin_in_top_org or h.check_access('sysadmin') else False}
+
```

Comment:
Why don't you use Group.get_top_level_groups  and has_user_permission_for_group_or_org or has_user_permission_for_some_org from ckan.lib.authz instead? In will be less complex and faster